kernel/src/color.rs

```diff
@@ -0,0 +1,54 @@
+use core::convert::TryFrom;
+
+/// Represents generic flower colors
+#[allow(dead_code)] // dead variants for completeness
+#[derive(Debug, Copy, Clone)]
+#[repr(u8)]
+pub enum Color {
```

Comment:
Should add a note on this being based off VGA colors. Perhaps in future this could be changed? What are thoughts on this? We could perhaps also have a conversion from rich colour (e.g rgba) to whatever is supported.

Reply:
Yeah, I had originally planned to make a generic for terminal colors / terminal characters but didn't want to over-complicate things. What do you think?

Reply:
So, we discussed this a bit IRL and came to the conclusion that we will expand the color module sometime (probably moving it into another crate or something)